### PR TITLE
Add http lexer

### PR DESCRIPTION
**Pull request by arduanov**
Http lexer based on pygments HttpLexer

https://bitbucket.org/birkenfeld/pygments-main/src/7941677dc77d4f2bf0bbd6140ade85a9454b8b80/pygments/lexers/textfmts.py?at=default&fileviewer=file-view-default#textfmts.py-113

### CODE REVIEW

File: lexers/http.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	tokens := []*Token{
+		{Name, groups[1]},
+		{Text, groups[2]},
+		{Operator, groups[3]},
+		{Text, groups[4]},
+		{Literal, groups[5]},
+		{Text, groups[6]},
+	}
+	iterators = append(iterators, Literator(tokens...))
+	return Concaterator(iterators...)
```
Comment (alecthomas) [inline lexers/http.go]: I think you can remove `iterators` and replace this with just `return Literator(tokens...)`?

File: lexers/http.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
+package lexers
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"strings"
+)
+
+var httpBodyContentType string
+
+// Http lexer.
```
Comment (alecthomas) [inline lexers/http.go]: Mind renaming this to HTTP?

File: lexers/http.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
+package lexers
+
+import (
+	. "github.com/alecthomas/chroma" // nolint
+	"strings"
+)
+
+var httpBodyContentType string
```
Comment (alecthomas) [inline lexers/http.go]: Unfortunately this won't be concurrent safe. You'll have to create a custom lexer that wraps the regex lexer and `Lexer.Tokenise(...)`. Your returned custom iterator will have to contain this state.

This is actually relatively straightforward. You can see an example of a stateful iterator in `coalesce.go`.

**Comment by arduanov** [comment]
Thanks for review, I made changes.

File: lexers/http.go [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+				{
+					contentType = strings.TrimSpace(token.Value)
+					pos := strings.Index(contentType, ";")
+					if pos > 0 {
+						contentType = strings.TrimSpace(contentType[:pos])
+					}
+				}
+			case token.Type == Generic && contentType != "":
+				{
+					lexer := MatchMimeType(contentType)
```
Comment (alecthomas) [inline lexers/http.go]: This is awesome!

**Comment by alecthomas** [comment]
This is really really cool, thank you!

**Pull request merged**